src/tv2-common/videoSwitchers/TriCaster.ts

```diff
@@ -175,17 +191,79 @@ export class TriCaster extends VideoSwitcherImpl {
 		)
 	}
 
-	public getDveTimelineObjects(_properties: DveProps): TSR.TSRTimelineObj[] {
-		throw new Error('Method not implemented.')
-	}
-
 	public updateUnpopulatedDveBoxes(
 		_timelineObject: TSR.TSRTimelineObj,
 		_input: number | SpecialInput
 	): TSR.TSRTimelineObj {
 		throw new Error('Method not implemented.')
 	}
 
+	public getDveTimelineObjects(dveProps: DveProps): TSR.TSRTimelineObj[] {
+		return [
+			literal<TSR.TimelineObjTriCasterME>({
+				id: '',
+				enable: dveProps.enable ?? { start: 0 },
+				layer: TRICASTER_LAYER_PREFIX + SwitcherDveLLayer.DveBoxes,
+				priority: 1,
+				content: {
+					deviceType: TSR.DeviceType.TRICASTER,
+					type: TSR.TimelineContentTypeTriCaster.ME,
+					me: literal<TSR.TriCasterMixEffectInEffectMode>({
+						transitionEffect: 8,
+						layers: this.generateDveBoxLayers(dveProps.content.boxes),
+						keyers: this.generateOverlayKeyer()
+					})
+				}
+			})
+		]
+	}
+
+	private generateDveBoxLayers(boxes: any[]): Partial<Record<TSR.TriCasterLayerName, TSR.TriCasterLayer>> {
+		return {
+			a: boxes[0].enabled ? this.generateDveBoxLayout(boxes[0]) : this.generateInvisibleBoxLayer(),
+			b: boxes[1].enabled ? this.generateDveBoxLayout(boxes[1]) : this.generateInvisibleBoxLayer(),
+			c: boxes[2].enabled ? this.generateDveBoxLayout(boxes[2]) : this.generateInvisibleBoxLayer(),
+			d: boxes[3].enabled ? this.generateDveBoxLayout(boxes[3]) : this.generateInvisibleBoxLayer()
+		}
+	}
+
+	private generateDveBoxLayout(box: any): TSR.TriCasterLayer {
+		return {
+			input: this.getInputName(box.source),
+			positioningAndCropEnabled: true,
+			position: this.dveConverter.convertPosition(box.x, box.y),
+			scale: this.dveConverter.convertScale(box.size),
+			crop: this.dveConverter.convertCrop(box)
+		}
+	}
+
+	private generateInvisibleBoxLayer(): TSR.TriCasterLayer {
+		return {
+			input: 'Black',
+			positioningAndCropEnabled: true,
+			position: {
+				x: -3.555,
+				y: -2
+			},
+			crop: {
+				down: 0,
+				up: 0,
+				left: 0,
+				right: 0
+			}
+		}
+	}
+
+	private generateOverlayKeyer(): Record<TSR.TriCasterKeyerName, TSR.TriCasterKeyer> {
+		return {
+			dsk1: {
+				input: this.getInputName(this.config.studio.SwitcherSource?.SplitArtFill ?? DVE_OVERLAY_INPUT_NUMBER),
```

Comment:
Since DveProps already contain a property with this value, I would prefer this to be consistent and use that instead. That way you can also remove the fallback

Reply:
It's been updated